shared/api/v3/course.ts

```diff
@@ -0,0 +1,160 @@
+import * as APIv4 from "../v4";
+
+type integer = number;
+
+export interface Course {
+    courseCode: string;
+    courseName: string;
+    courseSortKey: string[];
+    courseMutualExclusionKey: string[];
+    courseDescription: string;
+    courseInstructors: string[];
+    courseTerm: string;
+    courseSchedule: Schedule[];
+    courseCredits: number;
+    courseSeatsTotal: integer;
+    courseSeatsFilled: integer;
+    courseWaitlistLength: null;
+    courseEnrollmentStatus: string;
+    permCount: integer;
+}
+
+export interface Schedule {
+    scheduleDays: string;
+    scheduleStartTime: string;
+    scheduleEndTime: string;
+    scheduleStartDate: string;
+    scheduleEndDate: string;
+    scheduleTermCount: integer;
+    scheduleTerms: integer[];
+    scheduleLocation: string;
+}
```

Comment:
it would make my life easier if you defined these types in [zod](https://github.com/colinhacks/zod):

```typescript
import * as Z from 'zod';

const Course = Z.object({
  courseCode: Z.string(),
  courseName: Z.string(),
  // ...
  courseCredits: Z.number(),
  courseSeatsTotal: Z.number().int(),
})
export type Course = Z.TypeOf<typeof Course>

const Schedule = Z.object({
  // ...
})
export type Schedule = Z.TypeOf<typeof Schedule>
```

since eventually we're going to need all API-served types defined in Zod anyway, for type-safety/validation.

Reply:
Adding on to that, we can also add numerical check to all the numbers. For example, we can set `courseSeatsTotal`, `permCount`, etc. to be `z.number().nonnegative()`. In fact, none of our numerical type should have negative numbers inside.